src/pages/OnboardingWork/BaseOnboardingWork.tsx

```diff
@@ -41,11 +41,17 @@ function BaseOnboardingWork({shouldUseNativeStyles, onboardingPurposeSelected}:
                 return;
             }
             const work = values.work.trim();
-            const {adminsChatReportID} = Policy.createWorkspace(undefined, true, work);
-            Welcome.setOnboardingAdminsChatReportID(adminsChatReportID);
+            if (!onboardingPolicyID) {
+                const {adminsChatReportID, policyID} = Policy.createWorkspace(undefined, true, work);
+                Welcome.setOnboardingAdminsChatReportID(adminsChatReportID);
+                Welcome.setOnboardingPolicyID(policyID);
+            }
+            // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
+            Policy.updateGeneralSettings(onboardingPolicyID as string, work);
```

Comment:
Why do we need to call `Policy.updateGeneralSettings()` in the initial condition where we haven't created the workspace yet? If we just create it with the right name, there's no need to do another API call to rename it, an we can wrap that in an `else { }` block, right?

I think it would also allow to remove the eslint suppression.

Reply:
All good, that's what reviews are for :)